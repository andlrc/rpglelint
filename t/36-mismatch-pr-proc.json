[
   {
      "column":10,
      "file":"t/36-mismatch-pr-proc.rpgle",
      "line":30,
      "message":"conflicting types for 'proc2'",
      "severity":"warning"
   },
   {
      "column":8,
      "file":"t/36-mismatch-pr-proc.rpgle",
      "line":8,
      "message":"previous declaration of 'proc2' was here",
      "severity":"note"
   },
   {
      "column":10,
      "file":"t/36-mismatch-pr-proc.rpgle",
      "line":37,
      "message":"conflicting types for 'proc3'",
      "severity":"warning"
   },
   {
      "column":8,
      "file":"t/36-mismatch-pr-proc.rpgle",
      "line":13,
      "message":"previous declaration of 'proc3' was here",
      "severity":"note"
   }
]
