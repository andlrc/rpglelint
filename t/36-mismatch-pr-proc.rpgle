// rpglelint: -Wparameter-mismatch

dcl-pr proc1 int(10);
  a int(10);
  b int( 10 );
end-pr;

dcl-pr proc2 int(10);
  a int(10);
  b varchar(32);
end-pr;

dcl-pr proc3 int(10);
  a int(10);
  b int(10) options(*nopass);
end-pr;

dcl-pr proc4 int(10);
  a int(10) value;
  b int(10) value options(*nopass);
end-pr;

dcl-proc proc1;
  dcl-pi *n int( 10 );
    b int(10);
    a int(10);
  end-pi;
end-proc;

dcl-proc proc2;
  dcl-pi *n int(10);
    x int(10);
    y int(10);
  end-pi;
end-proc;

dcl-proc proc3;
  dcl-pi *n int(10);
    x int(10) options(*nopass);
    y int(10);
  end-pi;
end-proc;

dcl-proc proc4;
  dcl-pi *n int(10);
    x int(10) value;
    b int(10) options(*nopass) value;
  end-pi;
end-proc;
