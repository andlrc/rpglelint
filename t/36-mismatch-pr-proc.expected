t/36-mismatch-pr-proc.rpgle:30:10: warning: conflicting types for 'proc2' [-Wparameter-mismatch]
 dcl-proc proc2;
          ^
t/36-mismatch-pr-proc.rpgle:8:8: note: previous declaration of 'proc2' was here
 dcl-pr proc2 int(10);
        ^
t/36-mismatch-pr-proc.rpgle:37:10: warning: conflicting types for 'proc3' [-Wparameter-mismatch]
 dcl-proc proc3;
          ^
t/36-mismatch-pr-proc.rpgle:13:8: note: previous declaration of 'proc3' was here
 dcl-pr proc3 int(10);
        ^
